### PR TITLE
[stable/nginx-ingress] Add flag for daemonset controllers so that hostPort can optionally be used

### PR DESCRIPTION
**Pull request by jmatsushita**
Reviving closed stale PR #1906 from @digivava address @mgoodness review comments.

### CODE REVIEW

**Comment by jmatsushita** [comment]
@mgoodness Hi there, I've addressed your comments on @digivava PR #2854 Let me know if there's anything else.

**Comment by viglesiasce** [comment]
@jmatsushita can you make the port configurable and do something like:

```yaml
hostPort:
  enabled: false
  httpPort: 80
  httpsPort: 443
  statsPort: 18080
```

File: stable/nginx-ingress/templates/controller-daemonset.yaml [hunk 1]
```diff
@@ -87,13 +87,22 @@ spec:
           ports:
             - name: http
               containerPort: 80
+              {{- if .Values.controller.enablehostPort }}
+              hostPort: 80
+              {{- end }}
               protocol: TCP
             - name: https
               containerPort: 443
+              {{- if .Values.controller.enablehostPort }}
```
Comment (unguiculus) [inline stable/nginx-ingress/templates/controller-daemonset.yaml:96]: enablehostPort -> enableHostPort

File: stable/nginx-ingress/templates/controller-daemonset.yaml [hunk 2]
```diff
@@ -87,13 +87,22 @@ spec:
           ports:
             - name: http
               containerPort: 80
+              {{- if .Values.controller.enablehostPort }}
```
Comment (unguiculus) [inline stable/nginx-ingress/templates/controller-daemonset.yaml:90]: enablehostPort -> enableHostPort

File: stable/nginx-ingress/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: nginx-ingress
-version: 0.8.16
+version: 0.8.17
```
Comment (unguiculus) [inline stable/nginx-ingress/Chart.yaml:2]: Please bump minor version, since it's a new feature.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please update readme.

**Comment by unguiculus** [comment]
Also, consider @viglesiasce's comment. Otherwise, please close the PR if you're not interesting in it any more.

**Pull request closed without merging**